Added documentation note for `add_custom_type()`

addmix opened the pull request:
<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->
Fix for #72153

Calinou committed:
Fix Editor Description tooltip not changing until scene is reopened

You still need to select another node to see the tooltip updated first,
but you no longer need to save and reopen the scene to view the updated
tooltip.

raulsntos committed:
C#: Add `Skew` to `Transform2D` and fix `InterpolateWith`

- Add `Skew` property to `Transform2D`.
- Fix `InterpolateWith` in `Transform2D` to support skewed transforms.

Commit:
Add a theme usability setting which updates the touch area of UI elements (e.g: scrollbar) for the editor on touchscreen devices

Faless committed:
[NET] Refactor TLS configuration.

Use a TLSOptions configuration object which is created via static
functions.

- "TLSOptions.client": uses the standard CA and common name verification.
- "TLSOptions.client_unsafe": uses optional CA verification (i.e. if specified)
- "TLSOptions.server": is the standard server configuration (chain + key)

This will allow us to expand the TLS configuration options to include
e.g. mutual authentication without bloating the classes that uses
StreamPeerTLS and PacketPeerDTLS as underlying peers.

adamscott committed:
Fix `Resource::duplicate()` missing packed arrays

poohcom1 committed:
Fix autocomplete filter not including substrings

bruvzg committed:
Fix unsafe murmur3 hash use for the default material keys, expose alpha hash transparency mode for default materials and Label3D and Sprite3D.

bruvzg committed:
Move Array:set_typed to internal GDExtension structure and unexposed it.

reduz committed:
Refactor high quality texture import

* Only two texture import modes for low/high quality now:
  * S3TC/BPTC
  * ETC2/ASTC
* Makes sense given this is the general preferred and most compatible combination in most platforms.
* Removed lossy_quality from VRAM texture compression options. It was unused everywhere.
* Added a new "high_quality" option to texture import. When enabled, it uses BPTC/ASTC (BC7/ASTC4x4) instead of S3TC/ETC2 (DXT1-5/ETC2,ETCA).
* Changed MacOS export settings so required texture formats depend on the architecture selected.

This solves the following problems:

* Makes it simpler to import textures as high quality, without having to worry about the specific format used.
* As the editor can now run on platforms such as web, Mac OS with Apple Silicion and Android, it should no longer be assumed that S3TC/BPTC is available by default for it.

AThousandShips committed:
Hide clip_children for non-drawn CanvasItems

dalexeev committed:
GDScript: Fix broken export group annotations

bruvzg committed:
[RichTextLabel] Fix thread unsafe `set_physics_process_internal` usage. Use `WorkerThreadPool` instead of creating new threads.

ARez2 committed:
Add Unit test for InputEventMouse

Faless committed:
[Net] Remove StreamPeerTLS.blocking_handshake option.

Which was unused internally, and can be replaced by:

```
while tls.get_status() == tls.STATUS_HANDSHAKING:
  tls.poll()
```

aaronfranke committed:
Add support for interpolating skewed transforms

clayjohn committed:
Remove cap on number of items drawn in frame in 2D gl_compatibility render

Also clean up some names to make the overall organization more clear

Also remove cap on items per batch

addmix committed:
Added documentation note for `add_custom_type()`

addmix committed:
Added documentation note for `add_custom_type()`

timothyqiu committed:
Allow unicode identifier in GDScript syntax highlighter

dalexeev committed:
Improve stroke drawing on 2D collision shapes

akien-mga commented on doc/classes/EditorPlugin.xml:
```diff
@@ -409,6 +409,7 @@
 			<description>
 				Adds a custom type, which will appear in the list of nodes or resources. An icon can be optionally passed.
 				When a given node or resource is selected, the base type will be instantiated (e.g. "Node3D", "Control", "Resource"), then the script will be loaded and set to this object.
+				[b]Note:[/b] The base type is the base engine class which this type's class heirarchy inherits, not any custom type parent classes.
```
```suggestion
				[b]Note:[/b] The base type is the base engine class which this type's class hierarchy inherits, not any custom type parent classes.
```

dalexeev committed:
GDScript: Fix `@export_enum` works only with `int`

akien-mga committed:
Merge pull request #71896 from poohcom1/fix/autocomplete-substring-filter

Fix autocomplete filter not including substrings

akien-mga committed:
Merge pull request #72295 from timothyqiu/unicode-highlighter

Allow unicode identifier in GDScript syntax highlighter

akien-mga committed:
Merge pull request #71822 from adamscott/fix-resource-duplicate

Fix `Resource::duplicate()` missing packed arrays

akien-mga committed:
Merge pull request #72291 from clayjohn/GL-item-cap

Remove cap on number of items drawn in frame in 2D gl_compatibility renderer

akien-mga committed:
Merge pull request #72287 from aaronfranke/tr2d-interp

Add support for interpolating skewed Transform2Ds

Geometror committed:
[Noise/NoiseTexture2D] Allow disabling normalization

KoBeWi committed:
Improve Connection Dialog method list

TokageItLab committed:
Remove the max input limit & cleanup AnimationNodeTransition API

KoBeWi committed:
Tweak overrun behavior of EditorPropertyObjectID

raulsntos committed:
C#: Fix `Rotated` and `RotatedLocal`

Implementation was interchanged.

addmix committed:
Update doc/classes/EditorPlugin.xml

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

addmix committed:
Update doc/classes/EditorPlugin.xml

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

TokageItLab committed:
Tweak the name for duplicated animations in the editor

Vilcrow committed:
Removed declarations of removed methods of the EditorPropertyRevert class.

The implementation of these methods was removed in 1806ec7c14a8f038c0ff33a88d19087225c8d7ab.

TokageItLab committed:
Add remove immutable tracks option to glTF importer

Co-authored-by: Lyuma <xn.lyuma@gmail.com>

smix8 committed:
Create default World navigation maps on demand only

Changes that the default navigation maps of World resources are only created on first use.

BastiaanOlij committed:
Fix SSAO/SSIl being applied to reflection probes

TokageItLab committed:
Fix scale tool with view plane is weird in 3D

bruvzg committed:
[X11] Add support for dead keys without active IME. Fix IME focus and cleanup.

akien-mga committed:
Merge pull request #72213 from ARez2/ARez2/master

Add Unit test for InputEventMouse

akien-mga committed:
Merge pull request #72175 from dalexeev/gds-fix-export-group-annotations

GDScript: Fix broken export group annotations

akien-mga committed:
Merge pull request #72310 from Geometror/allow-disabling-noise-normalization

[Noise/NoiseTexture2D] Allow disabling normalization

akien-mga committed:
Merge pull request #72302 from dalexeev/shape-stroke

Improve stroke drawing on 2D collision shapes

akien-mga committed:
Merge pull request #72315 from raulsntos/dotnet/transform2d-skew

C#: Add `Skew` to `Transform2D` and fix `InterpolateWith`

akien-mga committed:
Merge pull request #66266 from Calinou/editor-description-fix-tooltip-update

Fix Editor Description tooltip not changing until scene is reopened

akien-mga committed:
Merge pull request #72336 from TokageItLab/dupe-animation-name

Tweak the name for duplicated animations in the editor

akien-mga committed:
Merge pull request #72115 from AThousandShips/clip_children_fix

Hide `clip_children` for some non-drawn CanvasItems

akien-mga committed:
Merge pull request #72326 from TokageItLab/unlimit-transition

Remove the max input limit & cleanup `AnimationNodeTransition` API

akien-mga committed:
Merge pull request #71657 from m4gr3d/increase_scroll_bar_size_for_touchscreen_main

Update the size of the scrollbar for the editor on touchscreen devices

akien-mga committed:
Merge pull request #71964 from bruvzg/mat_key

Fix unsafe murmur3 hash use for the default material keys, expose alpha hash transparency mode for default materials and Label3D and Sprite3D.

akien-mga committed:
Merge pull request #72362 from TokageItLab/fix-scale-gizmo

Fix scale tool with view plane is weird in 3D

groud committed:
Warn when a TileMap layer is Y-sorted but not the TileMap node itself.

akien-mga committed:
Merge pull request #72342 from TokageItLab/immutabletrack

Add remove immutable tracks option to glTF importer

akien-mga committed:
Merge pull request #72365 from groud/warn_ysort_tilemap

Warn when a TileMap layer is Y-sorted but not the TileMap node itself.

akien-mga committed:
Merge pull request #71995 from Faless/net/4.x_tls_verify

[NET] Refactor TLS configuration.

akien-mga committed:
Merge pull request #71989 from bruvzg/set_typed_move

Move Array:set_typed to internal GDExtension structure and unexposed it.

akien-mga committed:
Merge pull request #72370 from bruvzg/dead_xim

[X11] Add support for dead keys without active IME. Fix IME focus and cleanup.

akien-mga committed:
Merge pull request #72325 from raulsntos/dotnet/fix-72321

C#: Fix `Rotated` and `RotatedLocal`

akien-mga committed:
Merge pull request #72344 from smix8/default_navmap_on_demand_only_4.x

Create default World navigation maps on demand only

akien-mga committed:
Fix various typos with codespell

And include #72377.

Co-authored-by: Wiktor Kocielski <withaust@gmail.com>

akien-mga committed:
Merge pull request #72317 from KoBeWi/run_over_here

Add overrun to EditorPropertyObjectID

akien-mga committed:
Merge pull request #72305 from dalexeev/gfs-fix-export-enum

GDScript: Fix `@export_enum` works only with `int`

akien-mga committed:
Merge pull request #72340 from Vilcrow/cleanup-editor-property-revert

Removes declarations of removed methods of the EditorPropertyRevert class

akien-mga commented:
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

akien-mga committed:
Merge pull request #72312 from KoBeWi/signature_move

Improve Connection Dialog method list

yedpodtrzitko committed:
docs: replace File with FileAccess

TokageItLab committed:
Allow seek() without assigned animation in AnimationPlayer

MewPurPur committed:
Fix error messages of {}Tweener.new()

TokageItLab committed:
Allow the Reset option of NodeTransition to be set for each Input

akien-mga committed:
Merge pull request #72380 from TokageItLab/noanim-seek

Allow `seek()` without assigned animation in `AnimationPlayer`

akien-mga committed:
Merge pull request #72381 from yedpodtrzitko/yed/update-fileaccess-docs

docs: replace `File` with `FileAccess`

akien-mga committed:
Merge pull request #72388 from TokageItLab/transition-reset-each

Allow the Reset option of NodeTransition to be set for each Input

vonagam committed:
GDScript: Fix vararg method calls with exact arguments

clayjohn committed:
Add a shader error when trying to using hint_normal_roughness_texture in the gl_compatibility renderer

vnen committed:
GDScript: Fix match branches return check on release

The check for existence of `return` only existed on debug builds for
match branches. This could lead on an invalid error after exporting. Now
this is checked on relase too, so it works the same as the editor.

dsnopek committed:
Only customize resources during export if there are any export plugins in use

akien-mga committed:
Merge pull request #72356 from BastiaanOlij/fix_ssao_ssil_probes

Fix SSAO/SSIl being applied to reflection probes

akien-mga committed:
Merge pull request #72393 from clayjohn/GLES3-normal-roughness-error

Add a shader error when trying to using hint_normal_roughness_texture in the gl_compatibility renderer

akien-mga committed:
Merge pull request #72390 from vonagam/fix-allowed-vararg-ptrcall

GDScript: Fix vararg method calls with exact arguments

akien-mga committed:
Merge pull request #72383 from MewPurPur/correct-tween-errors

Fix error messages for {}Tweener.new()

akien-mga committed:
Merge pull request #72400 from vnen/gdscript-match-release-consistency

GDScript: Fix match branches return check on release

akien-mga committed:
Merge pull request #72208 from bruvzg/rtl_fix_thr_crash

[RichTextLabel] Fix thread unsafe `set_physics_process_internal` usage. Use `WorkerThreadPool` instead of creating new threads.

akien-mga committed:
Merge pull request #72402 from dsnopek/openxr-action-map-export-regression

Only customize resources during export if there are any export plugins in use

akien-mga committed:
Merge pull request #72031 from reduz/change-high-quality-texture-import

Refactor high quality texture import

addmix committed:
Merge branch 'master' of https://github.com/addmix/godot

addmix commented:
Don't know what I did wrong, following the PR workflow messed up the PR pretty badly. I'll make a new one.
